src/data_provider/src/sysInfoLinux.cpp

```diff
@@ -355,6 +364,118 @@ nlohmann::json SysInfo::getNetworks() const
+
+    file.close();
+
+    return processInfo;
+}
+
+void findInodeMatch(char path[], std::vector<int64_t>& inodes, bool& matchInode, const nlohmann::json& JSONInput)
+{
+    // Read symbolic link files
+    char fdFileContent[KByte] {0};
+
+    if (readlink(path, fdFileContent, sizeof(fdFileContent)) > 0)
+    {
+        // Parse inode from symbolic link.
+        int openBracketPos = std::string(fdFileContent).find("[");
+        int closeBracketPos = std::string(fdFileContent).find("]");
+        std::string match = std::string(fdFileContent).substr(openBracketPos + 1, closeBracketPos - openBracketPos - 1);
+        bool exists{false};
+        int64_t matchN{0};
+
+        for (const auto& i : JSONInput)
+        {
+            std::stringstream ss;
+            ss << match;
+            ss >> matchN;
+
+            if (i.at("inode") == matchN)
+            {
+                exists = true;
+            }
+        }
+
+        if (exists)
+        {
+            matchInode = true;
+            inodes.push_back(matchN);
+        }
+    }
+}
+
+void parseProcFS(const char path[], bool contRecursion, nlohmann::json& JSONInput, bool& matchInode, std::vector<int64_t>& inodes)
+{
+    struct dirent* dirent = nullptr;
+    DirPtr dir(opendir(path));
+
+    if (nullptr != dir.get())
+    {
+        while (nullptr != (dirent = readdir(dir.get())))
+        {
+            if (dirent->d_name[0] != '.')
+            {
+                char subPath[KByte] {0};
+                snprintf(subPath, sizeof(subPath), "%s/%s", path, dirent->d_name);
+
+                struct stat attr;
+                stat(subPath, &attr);
+
+                if (S_ISDIR(attr.st_mode) && contRecursion)
+                {
+                    if (isNumber(dirent->d_name))
+                    {
+                        inodes.clear();
+                        matchInode = false;
+                        parseProcFS(subPath, true, JSONInput, matchInode, inodes);
+                    }
+
+                    if (!strcmp(dirent->d_name, "fd"))
+                    {
+                        parseProcFS(subPath, false, JSONInput, matchInode, inodes);
+                    }
+                }
+                else if (!strcmp(dirent->d_name, "stat"))
+                {
+                    if (matchInode)
+                    {
+                        for (size_t i = 0; i < JSONInput.size(); ++i)
+                        {
+                            if (std::find(inodes.begin(), inodes.end(), JSONInput[i].at("inode")) != inodes.end())
+                            {
+                                std::pair<std::string, std::string> processInfo = parseProcessInfo(subPath);
+                                JSONInput[i].at("pid") = processInfo.first;
+                                JSONInput[i].at("process") = processInfo.second;
+                            }
+                        }
```

Comment:
I think something like this would also work.

```c++
for (auto& it : JSONInput) {
    if (std::find(inodes.begin(), inodes.end(), it.at("inode")) != inodes.end())
    {
        const auto processInfo{parseProcessInfo(subPath)};
        it.at("pid") = processInfo.first;
        it.at("process") = processInfo.second;
    }
}
```

Reply:
Changed